client-config/src/main/java/com/palantir/conjure/java/client/config/HttpsProxies.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.client.config;
+
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+import java.net.SocketAddress;
+
+/**
+ * Utility class for creating and identifying HTTPS proxy instances.
+ * <p>
+ * This class provides a way to create proxy instances that represent HTTPS connections, as there is no native support
+ * for an HTTPS proxy type in the Java {@link Proxy} class. When using a proxy created with this class, consumers should
+ * use the provided {@link #isHttps(Proxy)} method to determine if the proxy should be connected to using HTTPS.
+ */
+public final class HttpsProxies {
+
+    public static Proxy create(InetSocketAddress address, boolean https) {
```

Comment:
I don't think we need the `boolean https` here, the utility class name already implies https: `HttpsProxies.create(address)` is clear and concise imo :-)

Reply:
Let's add a `Preconditions.checkNotNull(address, "address is required")` to guard against nulls, since this is public API

Reply:
Updated!

Reply:
We still need to drop the parameter:
```suggestion
    public static Proxy create(InetSocketAddress address) {
```

Reply:
Whoops, yes of course